werkit/compute/graph/_custom_type.py

```diff
@@ -0,0 +1,87 @@
+import typing as t
+from abc import ABC, abstractmethod
+
+JSONType = t.Union[str, int, float, bool, None, t.Dict[str, t.Any], t.List[t.Any]]
+
+CanonicalType = t.TypeVar("CanonicalType")
```

Comment:
It's mentioned in the comments on CustomType: a custom type has the idea of a canonical in-memory representation and a JSON representation. The canonical in-memory type it's also what you will get back after deserializing. For the implementer's convenience, the compute function may return something of a different type, as long as it can be coerced to the canonical representation.

For example, in our application, we may want to be able to return some kind of CrossSection type which would get coerced to a Measurement. Measurement would be the canonical type.

Reply:
OK, this is clear